src/data/repositories/AlertD2Repository.ts

```diff
@@ -0,0 +1,119 @@
+import { D2Api } from "@eyeseetea/d2-api/2.36";
+import { apiToFuture, FutureData } from "../api-futures";
+import {
+    RTSL_ZEBRA_ALERTS_DISEASE_TEA_ID,
+    RTSL_ZEBRA_ALERTS_EVENT_TYPE_TEA_ID,
+    RTSL_ZEBRA_ALERTS_NATIONAL_DISEASE_OUTBREAK_EVENT_ID_TEA_ID,
+    RTSL_ZEBRA_ALERTS_NATIONAL_INCIDENT_STATUS_TEA_ID,
+    RTSL_ZEBRA_ALERTS_PROGRAM_ID,
+    RTSL_ZEBRA_ORG_UNIT_ID,
+} from "./consts/DiseaseOutbreakConstants";
+import { AlertOptions, AlertRepository } from "../../domain/repositories/AlertRepository";
+import { Id } from "../../domain/entities/Ref";
+import _ from "../../domain/entities/generic/Collection";
+import { Future } from "../../domain/entities/generic/Future";
+import {
+    D2TrackerTrackedEntity,
+    TrackedEntitiesGetResponse,
+} from "@eyeseetea/d2-api/api/trackerTrackedEntities";
+import { Maybe } from "../../utils/ts-utils";
+import { DataSource } from "../../domain/entities/disease-outbreak-event/DiseaseOutbreakEvent";
+
+type Filter = {
+    id: Id;
+    value: Maybe<string>;
+};
+
+export class AlertD2Repository implements AlertRepository {
+    constructor(private api: D2Api) {}
+
+    updateAlerts(alertOptions: AlertOptions): FutureData<void> {
+        const { dataSource, eventId, hazardTypeCode, incidentStatus, suspectedDiseaseCode } =
+            alertOptions;
+        const filter = this.getAlertFilter(dataSource, suspectedDiseaseCode, hazardTypeCode);
+
+        return this.getTrackedEntitiesByTEACode(filter).flatMap(response => {
+            const alertsToMap = response.map(trackedEntity => ({
+                ...trackedEntity,
+                attributes: [
+                    {
+                        attribute: RTSL_ZEBRA_ALERTS_NATIONAL_DISEASE_OUTBREAK_EVENT_ID_TEA_ID,
+                        value: eventId,
+                    },
+                    {
+                        attribute: RTSL_ZEBRA_ALERTS_NATIONAL_INCIDENT_STATUS_TEA_ID,
+                        value: incidentStatus,
+                    },
+                ],
+            }));
+
+            if (alertsToMap.length === 0) return Future.success(undefined);
+
+            return apiToFuture(
+                this.api.tracker.post(
+                    { importStrategy: "UPDATE" },
+                    { trackedEntities: alertsToMap }
+                )
+            ).map(saveResponse => {
+                if (saveResponse.status === "ERROR")
+                    throw new Error("Error mapping disease outbreak event id to alert");
```

Comment:
Here also: Future.error(new Error("Error mapping disease outbreak event id to alert"));